### ISSUE TITLE
[consistent-type-exports] Improve docs

### ISSUE THREAD

**Issue by Josh-Cena** [typescript-eslint/typescript-eslint#4769]
### Suggested Changes

The docs look a little... wrong to me.

> This rule aims to standardize the use of type exports style across a codebase.
>
> Given a class `Button`, and an interface `ButtonProps`, examples of code:

But the actual example is missing.

Further down, the "correct" example for `fixMixedExportsWithInlineTypeSpecifier`:

```ts
export { Button } from 'some-library';
export type { ButtonProps } from 'some-library';
export { Button, type ButtonProps } from 'some-library';
```

Is obviously duplicate export. Is that intended?

Furthermore, it may be beneficial to point out in "When Not To Use It" that `isolatedModules` already covers this:

```
Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.
```

### Affected URL(s)

https://typescript-eslint.io/rules/consistent-type-exports

**Comment by Josh-Cena** [comment]
@kmin-jeong Sure, PRs are always welcome as long as no-one seems to be working on it :) Especially for low-effort ones like this
